crates/core/src/io.rs

```diff
@@ -1,32 +1,34 @@
-use std::sync::{Arc, RwLock};
-
-use wasmtime_wasi::preview2::pipe::WritePipe;
+use wasmtime_wasi::preview2::{pipe::MemoryOutputPipe, HostOutputStream};
 
 /// An in-memory stdio output buffer.
-#[derive(Default)]
-pub struct OutputBuffer(Arc<RwLock<Vec<u8>>>);
+#[derive(Clone)]
+pub struct OutputBuffer(MemoryOutputPipe);
 
 impl OutputBuffer {
     /// Takes the buffered output from this buffer.
-    pub fn take(&mut self) -> Vec<u8> {
-        std::mem::take(&mut *self.0.write().unwrap())
+    pub fn contents(&self) -> bytes::Bytes {
+        self.0.contents()
     }
 
-    pub(crate) fn writer(&self) -> WritePipe<Vec<u8>> {
-        WritePipe::from_shared(self.0.clone())
+    pub(crate) fn writer(&self) -> impl HostOutputStream {
+        self.0.clone()
+    }
+}
+
+impl Default for OutputBuffer {
+    fn default() -> Self {
+        Self(MemoryOutputPipe::new(1024 * 1024))
```

Comment:
I don't know what this _should_ be, but 1MB seems too small; if I'm reading this right it would cap the WAGI output size, right?

Reply:
I believe that's correct yeah. To get equivalent behavior as before you'd use `usize::MAX`, so that's a possibility too.

Reply:
Well it's the size of the buffer for a single write not for the entire instance, but I think we should set it to `usize::MAX` since that's what we effectively had before.

@lann this type doesn't seem to be used except for in tests. Would it make sense to just get rid of it?

Reply:
Oh sure enough. It used to be used by wagi but it looks like it was removed at some point. :shrug: 

I don't see it used anywhere else, so should be fine to remove.